Src/CSharpier/CommandLineFormatter.cs

```diff
@@ -0,0 +1,323 @@
+                        );
+                        Console.WriteLine(failure);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Interlocked.Increment(ref this.ExceptionsValidatingSource);
+                    Console.WriteLine(
+                        GetPath()
+                        + " - failed with exception during syntax tree validation"
+                        + Environment.NewLine
+                        + ex.Message
+                        + ex.StackTrace
+                    );
+                }
+            }
+
+            if (this.Check)
+            {
+                if (result.Code != fileReaderResult.FileContents)
+                {
+                    Console.WriteLine(GetPath() + " - was not formatted");
+                    Interlocked.Increment(ref this.UnformattedFiles);
+                }
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+            Interlocked.Increment(ref this.Files);
+
+            if (!this.Check)
+            {
+                // purposely avoid async here, that way the file completely writes if the process gets cancelled while running.
+                File.WriteAllText(file, result.Code, fileReaderResult.Encoding);
+            }
+        }
+
+        private void PrintResults()
+        {
+            Console.WriteLine(
+                PadToSize("total time: ", 80)
+                + ReversePad(Stopwatch.ElapsedMilliseconds + "ms")
+            );
+            PrintResultLine("Total files", Files);
+
+            if (Validate)
+            {
+                PrintResultLine(
+                    "Failed syntax tree validation",
+                    FailedSyntaxTreeValidation
+                );
+
+                PrintResultLine(
+                    "Threw exceptions while formatting",
+                    ExceptionsFormatting
+                );
+                PrintResultLine(
+                    "files that threw exceptions while validating syntax tree",
+                    ExceptionsValidatingSource
+                );
+            }
+
+            if (Check)
+            {
+                PrintResultLine(
+                    "files that were not formatted",
+                    UnformattedFiles
+                );
+            }
+        }
+
+        private int ReturnExitCode()
+        {
+            if (
+                (Check && UnformattedFiles > 0)
+                || FailedSyntaxTreeValidation > 0
+                || ExceptionsFormatting > 0
+                || ExceptionsValidatingSource > 0
+            ) {
+                return 1;
+            }
+
+            return 0;
+        }
+
+        private static void PrintResultLine(string message, int count)
+        {
+            Console.WriteLine(
+                PadToSize(message + ": ", 80) + ReversePad(count + "  ")
+            );
+        }
+
+        private bool IgnoreFile(string filePath)
```

Comment:
This whole thing could be much simpler if you used a [Matcher](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.filesystemglobbing.matcher?view=dotnet-plat-ext-5.0)

You can construct one, add `**/*.cs` as an include, add your `**/*.g.cs` (etc. along with all the user specified ones) as excludes, and just call evaluate. It doesn't go to disk and expects you to provide the list of files, but that should be easy since you already have them all in memory already.